src/tools/cephfs/cephfs-shell

```diff
@@ -802,32 +804,31 @@ sub-directories, files')
                 dirs = []
                 for i in all_items:
                     for item in ls(path):
-                        d_name = item.d_name
+                        d_name = item.d_name.decode('utf-8')
                         if os.path.basename(i) == d_name:
                             if item.is_dir():
                                 dirs.append(os.path.join(path, d_name))
                 paths.extend(dirs)
                 continue
             else:
                 is_pattern = False
-            path = ''
-            path = os.path.normpath(os.path.join(
-                cephfs.getcwd().decode('utf-8'), path))
+            path = os.path.normpath(os.path.join(cephfs.getcwd().decode(
+                                    'utf-8'), path))
             if args.parent:
-                files = reversed(
-                    sorted(set(dirwalk(path))))
-                for path in files:
-                    path = os.path.normpath(path)
-                    if path[1:] != path:
+                files = reversed(sorted(set(dirwalk(path))))
+                for filepath in files:
+                    filepath = os.path.normpath(path)
+                    if filepath[1:] != path:
                         try:
-                            cephfs.rmdir(path)
+                            cephfs.rmdir(filepath)
```

Comment:
`filepath` needs to be converted to `bytes`.

Reply:
The conversion is not required. It was actually removed with this commit: https://github.com/ceph/ceph/commit/ba9f0eefa5466fd0e4bcac03582e8a538a897c26

Reply:
Ok. Oddly when I tested this PR last time I did find an error message produced for this line complaining about string-bytes conversion. But I don't see it now. Perhaps you rebased the PR branch.

Reply:
Anyways, I don't see the error message produced so this change is definitely not required.